snapcraft/meta/snap_yaml.py

```diff
@@ -18,27 +18,58 @@
 
 import textwrap
 from pathlib import Path
-from typing import Any, Dict, List, Optional, cast
+from typing import Any, Dict, List, Optional, Union, cast
 
 import yaml
 from pydantic_yaml import YamlModel
 
 from snapcraft.projects import Project
 
 
+class Socket(YamlModel):
+    """snap.yaml app socket entry."""
+
+    listen_stream: Union[int, str]
+    socket_mode: Optional[int]
+
+
 class SnapApp(YamlModel):
     """Snap.yaml app entry.
 
     This is currently a partial implementation, see
     https://snapcraft.io/docs/snap-format for details.
 
-    TODO: add missing properties, improve validation (CRAFT-802)
+    TODO: implement desktop (CRAFT-804)
+    TODO: implement extensions (CRAFT-805)
+    TODO: implement passthrough (CRAFT-854)
+    TODO: implement slots (CRAFT-816)
     """
 
     command: str
-    command_chain: List[str]
-    environment: Optional[Dict[str, Any]]
+    autostart: Optional[str]
```

Comment:
Struggling with pylint:
```
************* Module snapcraft.commands.__init__
snapcraft/commands/__init__.py:1:0: R0801: Similar lines in 2 files
==snapcraft.meta.snap_yaml:[51:60]
==snapcraft.projects:[86:95]
    completer: Optional[str]
    stop_command: Optional[str]
    post_stop_command: Optional[str]
    start_timeout: Optional[str]
    stop_timeout: Optional[str]
    watchdog_timeout: Optional[str]
    reload_command: Optional[str]
    restart_delay: Optional[str]
    timer: Optional[str] (duplicate-code)
```

It's not duplicate code, yet I can't find a good way to disable the `duplicate-code` warning in just these files.  
`# pylint: disable=duplicate-code` isn't supported (https://github.com/PyCQA/pylint/issues/214).

Reply:
Silenced with https://github.com/mr-cal/snapcraft/blob/097b7f04c8dc38d5e206f39c3b066f614b2671db/pyproject.toml#L48